src/osp/Resource/load_tinygltf.cpp

```diff
@@ -350,3 +350,119 @@ ResId osp::load_tinygltf_file(std::string_view filepath, Resources &rResources,
+        OSP_LOG_WARN("* has ImporterData: {}", pImportData != nullptr);
+        OSP_LOG_WARN("* has TinyGltf Extras: {}", pNodeExtras != nullptr);
+        return;
+    }
+
+    if (pImportData->m_scnTopLevel.ids_count() == 0)
+    {
+        OSP_LOG_WARN("Resource {} (gc_importer #{}) has no scenes!");
+        return;
+    }
+
+    auto const topLevelSpan = pImportData->m_scnTopLevel[0];
+
+    auto &rPrefabs = rResources.data_add<Prefabs>(restypes::gc_importer, importer);
+
+    int const objCount = pImportData->m_objParents.size();
+    rPrefabs.m_objMass      .resize(objCount, 0.0f);
+    rPrefabs.m_objShape     .resize(objCount, phys::EShape::None);
+
+    rPrefabs.m_prefabs      .data_reserve(objCount);
+    rPrefabs.m_prefabs      .ids_reserve(topLevelSpan.size());
+
+    rPrefabs.m_prefabParents.data_reserve(objCount);
+    rPrefabs.m_prefabParents.ids_reserve(topLevelSpan.size());
+
+    rPrefabs.m_prefabNames  .reserve(topLevelSpan.size());
+
+    // OSP parts are specified as top-level gltf nodes on the first scene
+    // with a name that starts with "part_"
+    // these rules may change
+
+    std::vector<int> prefabObjs;
+    std::vector<int> prefabParents;
+    prefabObjs.reserve(objCount);
+    prefabParents.reserve(objCount);
+
+    auto const process_obj_recurse = [&prefabObjs, &prefabParents, &rPrefabs, &rNodeExtras = *pNodeExtras, &rImportData = *pImportData] (auto&& self, int obj, int parent) -> void
+    {
+        auto const &name = rImportData.m_objNames[obj];
+        tinygltf::Value const &extras = rNodeExtras[obj];
+
+        if (extras.IsObject())
+        {
+            if (name.hasPrefix("col_"))
+            {
+                // is Collider
+                std::string_view const shapeName = extras.Get("shape").Get<std::string>();
+
+                rPrefabs.m_objShape[obj] = shape_from_name(shapeName);
+            }
+
+
+            if (tinygltf::Value massValue = extras.Get("massdry");
+                massValue.IsNumber())
+            {
+                rPrefabs.m_objMass[obj] = massValue.GetNumberAsDouble();
+            }
+        }
+
+        int const prefabIndex = prefabParents.size();
+        prefabParents.push_back(parent);
+        prefabObjs.push_back(obj);
+
+        // recurse into children
+        for (int child : rImportData.m_objChildren[obj])
+        {
+            self(self, child, prefabIndex);
+        }
+    };
+
+    int prefabIdNext = 0;
+
+    for (int const obj : topLevelSpan)
+    {
+        auto const &name = pImportData->m_objNames[obj];
+        if ( ! name.hasPrefix("part_"))
+        {
+            continue;
+        }
+
+        rPrefabs.m_prefabNames.emplace_back(name.exceptPrefix("part_"));
+
+        // Read children and populate prefabObjs and prefabParents
+        process_obj_recurse(process_obj_recurse, obj, -1);
+
+        rPrefabs.m_prefabs.emplace(
+                prefabIdNext, std::begin(prefabObjs), std::end(prefabObjs));
```

Comment:
std::move_iterator

Reply:
I don't believe this is any helpful when dealing with integer types?

Reply:
ah, yea, it wouldn't be.